sylt-compiler/src/compiler.rs

```diff
@@ -1191,6 +1202,7 @@ impl Compiler {
         }
     }
 
+    // TODO(ed): This should probably be cleaned up and moves out of the compiler?
```

Comment:
Yeah, or at least something has to be done to it. What, I think we should discuss.